database/user.go

```diff
@@ -30,13 +31,67 @@ const (
 	TierPremium80
 	// TierMaxReserved is a guard value that helps us validate tier values.
 	TierMaxReserved
+
+	// mbpsToBytesPerSecond is a multiplier to get from megabits per second to
+	// bytes per second.
+	mbpsToBytesPerSecond = 1024 * 1024 / 8
+
+	// filesAllowedPerTB defines a limit of number of uploaded files we impose
+	// on users. While we define it per TB, we impose it based on their entire
+	// quota, so an Extreme user will be able to upload up to 400_000 files
+	// before being hit with a speed limit.
```

Comment:
Should this be...?
```suggestion
	// before being hit with a storage limit.
```

Reply:
This is intended. Once you exceed your quota you get hit with a *speed* limit, i.e. all of your speed limitations become the same as an anonymous account.